File: app/src/main/java/com/autoai/readnotification/ui/MainActivity.java
package com.autoai.readnotification.ui;

import android.content.ContentResolver;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Build;
import android.provider.Settings;
import android.support.annotation.Nullable;
import android.support.annotation.RequiresApi;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.autoai.readnotification.MyNotifiService;
import com.autoai.readnotification.R;

public class MainActivity extends AppCompatActivity {

   // EditText editText;
    private static final String ACTION_NOTIFICATION_LISTENER_SETTINGS = "android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS";

     @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


         Button button2 = findViewById(R.id.button2);

        Intent intent = new Intent(MainActivity.this, MyNotifiService.class);
        startService(intent);//Start service
        final SharedPreferences sp = getSharedPreferences("msg", MODE_PRIVATE);

        button2.setOnClickListener(new View.OnClickListener() {
            @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP_MR1)
            @Override
            public void onClick(View v) {
                //Open listener reference message // Notification access
                Intent intent_s = new Intent(Settings.ACTION_NOTIFICATION_LISTENER_SETTINGS);
                startActivity(intent_s);
            }
        });
         if (isNotificationServiceRunning()) {
             Intent intent1 = new Intent(getBaseContext(),CustomizeMessageActivity.class);
             startActivity(intent1);
         }


    }


    @Override
    protected void onStart() {
        super.onStart();

    }

    public void openNotificationSettings(View view) {
        startActivity(new Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS));
    }
    private boolean isNotificationServiceRunning() {
         ContentResolver contentResolver = getContentResolver();
        String enabledNotificationListeners = Settings.Secure.getString(contentResolver, "enabled_notification_listeners");
        String packageName = getPackageName();
        return enabledNotificationListeners != null && enabledNotificationListeners.contains(packageName);
     }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (isNotificationServiceRunning()) {
            Intent intent = new Intent(getBaseContext(),CustomizeMessageActivity.class);
            startActivity(intent);
        }
    }
}

File: app/src/main/java/com/autoai/readnotification/ui/MainActivity.kt
package com.autoai.readnotification.ui

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.autoai.readnotification.MyNotifiService
import com.autoai.readnotification.R
import com.autoai.readnotification.ui.CustomizeMessageActivity
import com.autoai.readnotification.ui.fragments.AboutPopupFragment
import com.autoai.readnotification.ui.fragments.CustomizeMessageFragment
import com.autoai.readnotification.ui.fragments.NotificationListFragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
    lateinit var bottomNavigation: BottomNavigationView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            val fragment = CustomizeMessageFragment()
            supportFragmentManager.beginTransaction().replace(R.id.container, fragment, fragment.javaClass.getSimpleName())
                    .commit()
        }

        val intent = Intent(this@MainActivity, MyNotifiService::class.java)
        startService(intent) //Start service
        val sp = getSharedPreferences("msg", Context.MODE_PRIVATE)
      /*  button2.setOnClickListener { //Open listener reference message // Notification access
            val intent_s = Intent(Settings.ACTION_NOTIFICATION_LISTENER_SETTINGS)
            startActivity(intent_s)
        }*/
        if (isNotificationServiceRunning) {
           // val intent1 = Intent(baseContext, CustomizeMessageActivity::class.java)
           // startActivity(intent1)
        }

        bottom_navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)

    }

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { menuItem ->
        when (menuItem.itemId) {
            R.id.navigation_home -> {
                val fragment = CustomizeMessageFragment()
                supportFragmentManager.beginTransaction().replace(R.id.container, fragment, fragment.javaClass.getSimpleName())
                        .commit()
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_sms -> {
                val fragment = NotificationListFragment()
                supportFragmentManager.beginTransaction().replace(R.id.container, fragment, fragment.javaClass.getSimpleName())
                        .commit()
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_notifications -> {
                val fragment = AboutPopupFragment()
                supportFragmentManager.beginTransaction().replace(R.id.container, fragment, fragment.javaClass.getSimpleName())
                        .commit()
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onStart() {
        super.onStart()
    }

    fun openNotificationSettings(view: View?) {
        startActivity(Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS))
    }

    private val isNotificationServiceRunning: Boolean
        private get() {
            val contentResolver = contentResolver
            val enabledNotificationListeners = Settings.Secure.getString(contentResolver, "enabled_notification_listeners")
            val packageName = packageName
            return enabledNotificationListeners != null && enabledNotificationListeners.contains(packageName)
        }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (isNotificationServiceRunning) {
          //  val intent = Intent(baseContext, CustomizeMessageActivity::class.java)
           // startActivity(intent)
        }
    }

    companion object {
        // EditText editText;
        private const val ACTION_NOTIFICATION_LISTENER_SETTINGS = "android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS"
    }
}